router/connection_maker.go

```diff
@@ -14,12 +14,15 @@ const (
 	MaxInterval     = 10 * time.Minute
 )
 
+type PeerNormaliser func(string) string
+
 type ConnectionMaker struct {
 	ourself        *LocalPeer
 	peers          *Peers
 	targets        map[string]*Target
 	cmdLineAddress map[string]struct{}
 	actionChan     chan<- ConnectionMakerAction
+	peerNormaliser PeerNormaliser
```

Comment:
don't like the name. `normalisePeerAddr` is the best I could come up with. Yes, it's a bit long.